docarray/computation/abstract_comp_backend.py

```diff
@@ -104,14 +104,26 @@ def detach(tensor: 'TTensor') -> 'TTensor':
         """
         ...
 
+    @staticmethod
+    @abstractmethod
+    def dtype(tensor: 'TTensor'):
```

Comment:
I think it is fine to have it be `Any`, the different frameworks will have their own classes for this. Since the dtype of own framework should never be used by another framework I don't think we have to unify it.
I think in the subclasses (concrete comp backends) you can then specify the actual type